chime/view_functions.py

```diff
@@ -588,6 +587,92 @@ def make_activity_history(repo):
 
     return history
 
+def summarize_activity_history(repo=None, history=None, branch_name=u''):
+    ''' Make an object that summarizes an activity's history. If no history is passed, will get a new one from make_activity_history()
+
+        The object looks like this:
+        {
+            'summary': u'3 articles and 1 category have been changed',
+            'changes': [
+                {'edit_path': u'', 'display_type': u'Article', 'actions': u'Created, Edited, Deleted', 'title': u'How to Find Us'},
+                {'edit_path': u'/tree/34246e3/edit/contact/hours-of-operation/', 'display_type': u'Article', 'actions': u'Created, Edited', 'title': u'Hours of Operation'},
+                {'edit_path': u'/tree/34246e3/edit/contact/driving-directions/', 'display_type': u'Article', 'actions': u'Created, Edited', 'title': u'Driving Directions'},
+                {'edit_path': u'/tree/34246e3/edit/contact/', 'display_type': u'Category', 'actions': u'Created', 'title': u'Contact'}
+            ]
+        }
+    '''
+    # an empty summary object
+    summary = dict(summary=u'', changes=[])
+
+    # make sure we've got what we need
+    if not history:
+        try:
+            history = make_activity_history(repo=repo)
+        except:
+            # we weren't give what we need, return an empty summary
+            return summary
+
+    ed_lookup = {'create': u'created', 'edit': u'edited', 'delete': u'deleted'}
+    change_lookup = {}
+    display_types_encountered = []
+    for action in reversed(history):
+        # we only care about edits
+        if action['commit_category'] == u'edit':
```

Comment:
This is essentially a filter; can the stuff in the `if`-clause be moved up a level, and prefixed with something like this?

    history = [action for action in reversed(history) if action['commit_category'] == u'edit'`